hibernate-core/src/main/java/org/hibernate/cfg/Environment.java

```diff
@@ -177,28 +179,31 @@ public static void verifyProperties(Map<?,?> configurationValues) {
 		//Set USE_REFLECTION_OPTIMIZER to false to fix HHH-227
 		GLOBAL_PROPERTIES.setProperty( USE_REFLECTION_OPTIMIZER, Boolean.FALSE.toString() );
 
-		try {
-			InputStream stream = ConfigHelper.getResourceAsStream( "/hibernate.properties" );
+		List<ClassLoader> classLoaders = new ArrayList<>();
```

Comment:
I think we can specify the size here:
```
List<ClassLoader> classLoaders = new ArrayList<>( 2 );
```

Reply:
I agree, although I'd use varargs

Reply:
> I agree, although I'd use varargs

To clarify, I mean 

```
class ResourcesHelper {
    ...

    public static InputStream locateResourceAsStream(String resourceName, ClassLoader... classLoaders) { ... }
}
```

and then here:

```
final InputStream stream = ResourcesHelper.locateResourceAsStream( 
        "/hibernate.properties",
        Thread.currentThread().getContextClassLoader(),
        Environment.class.getClassLoader()
);
```

Reply:
> > I agree, although I'd use varargs
> 
> To clarify, I mean
> 
> ```
> class ResourcesHelper {
>     ...
> 
>     public static InputStream locateResourceAsStream(String resourceName, ClassLoader... classLoaders) { ... }
> }
> ```
> 
> and then here:
> 
> ```
> final InputStream stream = ResourcesHelper.locateResourceAsStream( 
>         "/hibernate.properties",
>         Thread.currentThread().getContextClassLoader(),
>         Environment.class.getClassLoader()
> );
> ```

Yes, I did get that, but thanks anyhow for the clarification ;)